src/main/java/com/google/sps/data/BackgroundTaskManager.java

```diff
@@ -26,8 +30,21 @@ public void replaceFaultyInstances() {
     }
 
     @Override
-    public void deleteInactiveAttendees() {
-        throw new RuntimeException("Unimplemented");
+    public void deleteInactiveAttendees(DatastoreClientInterface datastore, String sessionId) { 
+        List<Attendee> attendees = datastore.getAttendeesInSession(sessionId);
+        for(int i = 0; i <= attendees.length(); i++){
```

Comment:
I think this produces an IndexOutOfBoundsException. I think you want "i < attendees.size()"